### PR TITLE
Upgrade Go version to 1.19.4 and upgrade app dependencies

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Includes:
- Go version upgrade to 1.19.4 and updated CI files to reference docker images that use Go version 1.19.4
- Add golangci linters to lint target and fixup lint issues against go code
- dependency upgrades
  - log.go
  - dp-net/v2
  - dp-api-clients-go
  - dp-healthcheck
  

### How to review

- CI docker images locked down to specific versions instead of latest
- Check changes make sense

### Who can review

Anyone

### CODE REVIEW

File: ci/audit.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ image_resource:
   type: docker-image
   source:
     repository: onsdigital/dp-concourse-tools-nancy
-    tag: latest
+    tag: 1.19.4-rc.1
```
Comment (redhug1) [inline ci/audit.yml:8]: I don't think you should be using a release candidate
I've been using 1.19.2 elsewhere, which is good enough

File: ci/build.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: onsdigital/dp-concourse-tools-node-go
-    tag: latest
+    tag: 1.19.4-rc.1
```
Comment (redhug1) [inline ci/build.yml:9]: I don't think you should be using a release candidate
I've been using 1.19.2 elsewhere, which is good enough

File: ci/component.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: onsdigital/dp-concourse-tools-chrome-go
-    tag: latest
+    tag: 1.19.4-rc.1
```
Comment (redhug1) [inline ci/component.yml:9]: I don't think you should be using a release candidate
I've been using 1.19.2 elsewhere, which is good enough

File: ci/unit.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: onsdigital/dp-concourse-tools-node-go
-    tag: latest
+    tag: 1.19.4-rc.1
```
Comment (redhug1) [inline ci/unit.yml:9]: I don't think you should be using a release candidate
I've been using 1.19.2 elsewhere, which is good enough

File: ci/lint.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: latest
+    tag: 1.19.4
```
Comment (redhug1) [inline ci/lint.yml:9]: if you can make this 1.19.2 as per other suggestions, ta.

**Review by redhug1 — CHANGES_REQUESTED** [review]

File: ci/audit.yml (hunk 1 above)
Comment (nshumoogum) [reply ci/audit.yml:8]: The docker image is badly named, the docker image contains Go version 1.19.4 and a version of nancy; I'll update tag today; same goes for other yml files in ci folder

Comment (nshumoogum) [reply ci/audit.yml:8]: Also Go version 1.19.4 is stable Go version with security fixes; better to use latest (stable) version - https://go.dev/doc/devel/release#go1.19.minor

Comment (redhug1) [reply ci/audit.yml:8]: its the '-rc.1' part that always rings alarm bells ...

**Review by redhug1 — APPROVED** [review]
LGTM

**Pull request merged**